### PR TITLE
feat: keep `base64offset|contains` value as is

### LINKED ISSUE #313: Do not error on |base64offset|contains
@fukusuket 
Now that Hayabusa v2.2.2 supports `|base64offset|contains`, we will need to not produce errors when converting rules. Can you write an exception for `|base64offset|contains` in `hayabusa.py` so that we can add these rules?

### PR DESCRIPTION
**Pull request by fukusuket**
Although it is not a good method... 🤔 it was possible to deal with `base64offset|contains` by customizing the sigmac code(rule.py)

## What Changed

- Fixed #313
- Added `rule.py` (from Sigma repository : https://github.com/SigmaHQ/sigma/blob/master/tools/sigma/parser/rule.py)
   - changed the conversion process to keep `base64offset|contains` value as is
     - [This code block](https://github.com/Yamato-Security/hayabusa-rules/blob/313-skip-base64offset-contains-convertion/tools/sigmac/rule.py#L84-L86) is the difference from the original `rule.py`.
   - because this logic is only for the `Hayabusa rule` and cannot be made into a PR that can be applied in the `Sigma` repository
   - A workaround for [this sigmac behavior](https://github.com/Yamato-Security/hayabusa-rules/issues/313#issuecomment-1435977603)
- Update README and GitHub actions

### Environment 
- OS: macOS montery version 13.1
- Hard: Macbook Air(M1, 2020) , Memory 8GB, Core 8

### Test1
Conversion of a rule with the `base64offset|contains` modifier was possible as follows.
```
fukusuke@fukusukenoAir sigma % python3 ./tools/sigmac -t hayabusa -c ./tools/config/generic/windows-audit.yml -c ./tools/config/generic/windows-services.yml --defer-abort ./rules/windows/process_creation/proc_creation_win_powershell_base64_frombase64string.yml
title: PowerShell Base64 Encoded FromBase64String Keyword
ruletype: Sigma
author: Florian Roth (Nextron Systems)
date: 2019/08/24
description: Detects usage of a base64 encoded "FromBase64String" cmdlet in a process
    command line
detection:
    SELECTION_1:
        EventID: 4688
    SELECTION_2:
        Channel: Security
    SELECTION_3:
        CommandLine|base64offset|contains: ::FromBase64String
....
```

### Test2
Compared the results of `python3 convert.py diff` before/after fix.
I confirmed that rules related to `base64offset|contains` are only the following (`11 rules`)
```
fukusuke@fukusukenoAir sigma % diff -r hayabusa_rules hayabusa_rules_new
Only in hayabusa_rules_new/builtin/process_creation: proc_creation_win_powershell_base64_frombase64string.yml
Only in hayabusa_rules_new/builtin/process_creation: proc_creation_win_powershell_base64_iex.yml
Only in hayabusa_rules_new/builtin/process_creation: proc_creation_win_powershell_base64_mppreference.yml
Only in hayabusa_rules_new/builtin/process_creation: proc_creation_win_powershell_defender_disable_feature.yml
Only in hayabusa_rules_new/builtin/security: win_security_cobaltstrike_service_installs.yml
Only in hayabusa_rules_new/builtin/system: win_system_cobaltstrike_service_installs.yml
Only in hayabusa_rules_new/sysmon/process_creation: proc_creation_win_powershell_base64_frombase64string.yml
Only in hayabusa_rules_new/sysmon/process_creation: proc_creation_win_powershell_base64_iex.yml
Only in hayabusa_rules_new/sysmon/process_creation: proc_creation_win_powershell_base64_mppreference.yml
Only in hayabusa_rules_new/sysmon/process_creation: proc_creation_win_powershell_defender_disable_feature.yml
Only in hayabusa_rules_new/sysmon/wmi_event: sysmon_wmi_susp_encoded_scripts.yml
```

### Test3
Compared the results of `hayabusa csv-timeline -d ./hayabusa-sample-evtx -o out.csv` before/after fix.
I confirmed that 
 - the difference in the number of rules is `11`
 - no rule error
 - the detection count increase(critical/high)

before
```
Loading detections rules. Please wait.

Excluded rules: 15
Noisy rules: 7 (Disabled)

Experimental rules: 1706 (49.78%)
Stable rules: 218 (6.36%)
Test rules: 1503 (43.86%)

Hayabusa rules: 148
Sigma rules: 3279
Total enabled detection rules: 3427

Scanning in progress. Please wait.
...
Results Summary:

First Timestamp: 2009-07-14 13:56:45.074 +09:00
Last Timestamp: 2023-01-24 20:54:42.899 +09:00
Events with hits / Total events: 19,616 / 47,465 (Data reduction: 27,849 events (58.67%))

Total | Unique detections: 32,434 | 613
Total | Unique critical detections: 49 (0.15%) | 20 (3.26%)
Total | Unique high detections: 6,260 (19.30%) | 276 (45.02%)
Total | Unique medium detections: 1,652 (5.09%) | 190 (31.00%)
Total | Unique low detections: 6,343 (19.56%) | 74 (12.07%)
Total | Unique informational detections: 18,130 (55.90%) | 53 (8.65%)
```

after
```
Loading detections rules. Please wait.

Excluded rules: 15
Noisy rules: 7 (Disabled)

Experimental rules: 1708 (49.68%)
Stable rules: 218 (6.34%)
Test rules: 1512 (43.98%)

Hayabusa rules: 148
Sigma rules: 3290
Total enabled detection rules: 3438

Scanning in progress. Please wait.
...
Results Summary:

First Timestamp: 2009-07-14 13:56:45.074 +09:00
Last Timestamp: 2023-01-24 20:54:42.899 +09:00
Events with hits / Total events: 19,616 / 47,465 (Data reduction: 27,849 events (58.67%))

Total | Unique detections: 32,445 | 616
Total | Unique critical detections: 55 (0.17%) | 21 (3.41%)
Total | Unique high detections: 6,265 (19.31%) | 278 (45.13%)
Total | Unique medium detections: 1,652 (5.09%) | 190 (30.84%)
Total | Unique low detections: 6,343 (19.55%) | 74 (12.01%)
Total | Unique informational detections: 18,130 (55.88%) | 53 (8.60%)
```
### Test4
Compared the result of `hayabusa csv-timeline -d ./hayabusa-sample-evtx -o out.csv -p super-verbose` before/after fix.
I confirmed that the difference in the detection rule is only related to `base64offset` with the command below.

```
./hayabusa csv-timeline -d ./hayabusa-sample-evtx -o before.csv -q -p super-verbose
... move rules folder
./hayabusa csv-timeline -d ./hayabusa-sample-evtx -o after.csv -q -p super-verbose
cat before.csv | awk -F"," '{print $11}' > before-rules.csv
cat after.csv | awk -F"," '{print $11}' > after-rules.csv
diff before-rules.csv after-rules.csv
2764a2765
> CobaltStrike Service Installations - System
16646a16648
> CobaltStrike Service Installations - System
16681a16684
> CobaltStrike Service Installations - System
16717a16721
> CobaltStrike Service Installations - System
16819a16824
> CobaltStrike Service Installations - System
17361a17367
> CobaltStrike Service Installations - System
22240a22247
> PowerShell Base64 Encoded IEX Keyword
22247a22255
> PowerShell Base64 Encoded IEX Keyword
24222a24231,24233
> CobaltStrike Service Installations - Security
> CobaltStrike Service Installations - Security
> CobaltStrike Service Installations - Security
```

I have confirmed that it works locally, but I have not confirmed that it works on Github action...
I would appreciate it if you could review🙏

### CODE REVIEW

**Comment by YamatoSecurity** [comment]
I confirmed that the `base64offset|contains` rules were converted without problem in the action workflow and they generate 11 parsing errors on version 2.2.0 but no parsing errors on 2.2.2 so LGTM!
We are going to try to fix this in the upstream sigma repository first and if that doesn't work let's merge this.

**Review by YamatoSecurity — APPROVED** [review]
@fukusuket When we talked directly, I said that we might as well wait until things get fixed upstream but let's merge this as it is now so that users can use the base64offset rules as they add some good rules. So no hurry on getting things fixed upstream, etc...

**Comment by fukusuket** [comment]
@YamatoSecurity  I've got it! Thank you for your confirmation :) I would appreciate it if you could merge it💪

**Pull request merged**
